### PR TITLE
CEDS-1907 Validation for country duplication and countries limit

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/controllers/declaration/RoutingCountriesController.scala [hunk 1]
```diff
@@ -45,9 +45,7 @@ class RoutingCountriesController @Inject()(
     extends FrontendController(mcc) with I18nSupport with ModelCacheable {
 
   def displayRoutingQuestion(mode: Mode, fastForward: Boolean): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
-    val containsRoutingCountries: Boolean = request.cacheModel.locations.routingCountries.nonEmpty
-
-    if (fastForward && containsRoutingCountries) {
+    if (fastForward && request.cacheModel.containRoutingCountries) {
```
Comment (bambuchaAdm) [inline app/controllers/declaration/RoutingCountriesController.scala]: maybe adding () to `containRoutingCountries` would increase readbility - in first moment i assume that is one of declaration field

File: app/controllers/declaration/RoutingCountriesSummaryController.scala [hunk 2]
```diff
@@ -67,13 +68,20 @@ class RoutingCountriesSummaryController @Inject()(
...
+
+          validatedForm.fold(formWithErrors => BadRequest(routingCountriesSummaryPage(mode, formWithErrors, countries)), _ => {
+            redirectFromSummaryPage(mode, validAnswer)
+          })
+        }
       )
   }
 
+  private def redirectFromSummaryPage(mode: Mode, answer: Boolean)(implicit request: JourneyRequest[AnyContent]): Result =
+    if (answer) navigator.continueTo(controllers.declaration.routes.RoutingCountriesController.displayRoutingCountry(mode))
```
Comment (bambuchaAdm) [inline app/controllers/declaration/RoutingCountriesSummaryController.scala]: please add braces - now it hard to read

File: app/forms/declaration/RoutingQuestionYesNo.scala [hunk 3]
```diff
@@ -33,10 +33,11 @@ object RoutingQuestionYesNo {
 
   val allowedValues: Seq[String] = Seq(yes, no)
 
-  def form(): Form[Boolean] = Form(
+  def form(cachedCountries: Seq[String] = Seq.empty): Form[Boolean] = Form(
     Forms.mapping(
       "answer" -> requiredRadio("declaration.routingQuestion.empty")
         .verifying("declaration.routingQuestion.error", isContainedIn(allowedValues))
+        .verifying(s"declaration.routingCountries.limit", _ => cachedCountries.length < DestinationCountries.limit)
```
Comment (bambuchaAdm) [inline app/forms/declaration/RoutingQuestionYesNo.scala]: if you reach 99 containers you won't be able to submit "No" version

Comment (PatrykRudnicki) [reply app/forms/declaration/RoutingQuestionYesNo.scala]: Fixed, please check now

**Pull request merged**